stdlib/LinearAlgebra/test/structuredbroadcast.jl

```diff
@@ -146,4 +146,13 @@ end
     @test L .+ UnitL .+ UnitU .+ U .+ D == L + UnitL + UnitU + U + D
     @test L .+ U .+ D .+ D .+ D .+ D == L + U + D + D + D + D
 end
+@testset "Broadcast Returned Types" begin
+    # Issue 35245
+    Bu = Bidiagonal(ones(4), ones(3), :U)
+    Bl = Bidiagonal(ones(4), ones(3), :L)
+    @test typeof(Bu .+ Bl) == Tridiagonal{Float64,Array{Float64,1}}
+    @test typeof(Bl .+ Bu) == Tridiagonal{Float64,Array{Float64,1}}
+    @test typeof(Bu .+ Bu) == Bidiagonal{Float64,Array{Float64,1}}
+    @test typeof(Bl .+ Bl) == Bidiagonal{Float64,Array{Float64,1}}
+end
```

Comment:
Could you please add some tests that also check the numerical result, and maybe another operation? I'd suggest to use randomized (off-)diagonal vectors, to verify that the (off-)diagonals end up in the right position.

Reply:
Added more tests in cd76ffb for `.*` .  With d0f3ec4 all the Tridiagonals and Bidiagonals are created with rand.

Reply:
To play it safe, it would be good to have different super- and sub-diagonals... Otherwise, with `+` you can't distinguish which one is which.

Reply:
Made the super and sub diagonal arrays distinct in afc630b